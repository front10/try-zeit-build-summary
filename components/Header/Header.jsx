import React from 'react';
import PropTypes from 'prop-types';
import useStyles from '@front10/helpers/dist/UseStyles';

const Header = props => {
  const { tag, content, customHtml, id } = props;
  const Type = tag;
  const [className, style] = useStyles(props);

  return customHtml ? (
    <Type
      id={id}
      className={`fr-${tag} ${className}`}
      style={style}
      dangerouslySetInnerHTML={{ __html: customHtml }}
    />
  ) : (
    <Type id={id} className={`fr-${tag} ${className}`} style={style}>
      {content}
    </Type>
  );
};

Header.propTypes = {
  /**
   * Type of tag element. Can be <code>'h1'</code>, <code>'h2'</code>, <code>'h3'</code>, <code>'h4'</code>, <code>'h5'</code> and <code>'h6'</code>
   */
  tag: PropTypes.string,
  /**
   * Content of the component
   */
  content: PropTypes.string,
  /**
   * CustomHtml of the component
   */
  customHtml: PropTypes.string,
  id: PropTypes.string
};

Header.defaultProps = {
  tag: 'h2',
  content: '',
  customHtml: '',
  id: null
};
Header.componentName = 'Header';

export default Header;
