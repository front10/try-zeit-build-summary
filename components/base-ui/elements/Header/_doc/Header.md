# Header
  [![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
  

  ## Properties
  | </br>Name | </br>Type | </br>Summary | </br>Default | 
| ---- | ---- | ---- | ---- |
| content | `string` | Content of the component | `''` |
| customHtml | `string` | CustomHtml of the component | `''` |
| id | `string` |  | `null` |
| tag | `string` | Type of tag element. Can be <code>'h1'</code>, <code>'h2'</code>, <code>'h3'</code>, <code>'h4'</code>, <code>'h5'</code> and <code>'h6'</code> | `'h2'` |