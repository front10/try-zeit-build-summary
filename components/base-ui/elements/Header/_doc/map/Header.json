{"description":"","displayName":"Header","methods":[],"props":{"className":{"type":{"name":"string"},"required":false,"description":"Class name of header","defaultValue":{"value":"''","computed":false}},"style":{"type":{"name":"objectOf","value":{"name":"any"}},"required":false,"description":"Css style applied to the component","defaultValue":{"value":"null","computed":false}},"tag":{"type":{"name":"string"},"required":false,"description":"Type of tag element. Can be <code>'h1'</code>, <code>'h2'</code>, <code>'h3'</code>, <code>'h4'</code>, <code>'h5'</code> and <code>'h6'</code>","defaultValue":{"value":"'h2'","computed":false}},"content":{"type":{"name":"string"},"required":false,"description":"Content of the component","defaultValue":{"value":"''","computed":false}},"customHtml":{"type":{"name":"string"},"required":false,"description":"CustomHtml of the component","defaultValue":{"value":"''","computed":false}},"id":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"null","computed":false}}},"filename":"/Users/darielnoel/repositories/front10/airmodules/src/components/elements/Header/Header.jsx"}