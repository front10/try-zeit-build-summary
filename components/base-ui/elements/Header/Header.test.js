import React from 'react';
import { render } from 'test-utils';
import Header from './Header';


test('render all', () => {
  const { container } = render(<Header />);
  expect(container).toMatchSnapshot();
});
