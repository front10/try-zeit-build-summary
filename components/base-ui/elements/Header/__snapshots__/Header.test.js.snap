// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render all 1`] = `
<h2
  className="fr-h2 "
  id={null}
  style={
    Object {
      "filter": undefined,
    }
  }
>
  
</h2>
`;
